cherrypy/test/test_native.py

```diff
@@ -0,0 +1,35 @@
+"""Test the native server."""
+
+import pytest
+from requests_toolbelt import sessions
+
+import cherrypy._cpnative_server
+
+
+pytestmark = pytest.mark.skipif(
+    'sys.platform == "win32"',
+    reason='tests fail on Windows',
+)
+
+
+@pytest.fixture
+def cp_native_server(request):
+    """A native server."""
+    class Root(object):
+        @cherrypy.expose
+        def index(self):
+            return 'Hello World!'
+
+    cls = cherrypy._cpnative_server.CPHTTPServer
+    cherrypy.server.httpserver = cls(cherrypy.server)
+
+    cherrypy.tree.mount(Root(), '/')
+    cherrypy.engine.start()
+    request.addfinalizer(cherrypy.engine.stop)
+    url = 'http://localhost:{cherrypy.server.socket_port}'.format(**globals())
+    return sessions.BaseUrlSession(url)
+
+
+def test_basic_request(cp_native_server):
```

Comment:
@jaraco it looks like this test doesn't catch the bug for some reason.

Ref:
* https://github.com/cherrypy/cherrypy/issues/1377#issuecomment-496252972
* https://github.com/cherrypy/cherrypy/pull/1784#issuecomment-504704035